src/popup/components/SwapRoute.vue

```diff
@@ -30,78 +30,104 @@
         return [];
       }
-      const resolver = getTransactionTokenInfoResolver(camelCase(this.transaction.tx.function));
+      const resolver = getTransactionTokenInfoResolver(txFunction);
       if (!resolver) return [];
-      let { tokens } = resolver(this.transaction, this.availableTokens);
-      const index = this.transaction.tx.arguments.findIndex(({ type }) => type === 'list');
-      if (index >= 0 && this.transaction.tx.arguments[index].value.length > tokens.length) {
+      let { tokens } = resolver(props.transaction, availableTokens.value);
+      const index = props.transaction.tx.arguments.findIndex(({ type }) => type === 'list');
+
+      if (index >= 0 && props.transaction.tx.arguments[index].value.length > tokens.length) {
         tokens = [
           tokens[0],
-          ...this.transaction.tx.arguments[index].value
-            .slice(1, this.transaction.tx.arguments[index].value.length - 1)
-            .map(({ value }) => this.availableTokens[value]),
+          ...props.transaction.tx.arguments[index].value
+            .slice(1, props.transaction.tx.arguments[index].value.length - 1)
+            .map((element: any) => availableTokens.value[element.value]),
           tokens[1],
         ];
       }
-      const { nodeNetworkId } = useAeSdk({ store: this.$store });
-      const waeContract = DEX_CONTRACTS[nodeNetworkId.value]?.wae;
-      if (tokens[0].isAe && waeContract && !waeContract?.includes(tokens[1].contractId)) {
+
+      const waeContract = DEX_CONTRACTS[nodeNetworkId.value!]?.wae;
+
+      if (tokens[0].isAe && waeContract && !waeContract?.includes(tokens[1].contractId as any)) {
         tokens.unshift({
           ...tokens[0],
           isAe: true,
         });
         tokens[1].isAe = false;
       }
       if (tokens[tokens.length - 1].isAe && waeContract
-        && !waeContract?.includes(tokens[tokens.length - 2].contractId)) {
+        && !waeContract?.includes(
+            tokens[tokens.length - 2].contractId as Encoded.ContractAddress,
+        )
+      ) {
         tokens[tokens.length - 1].isAe = false;
         tokens.push({ ...tokens[tokens.length - 1], isAe: true });
       }
       return tokens;
-    },
-  },
-  methods: {
-    checkWaeAeTx(idx) {
-      if (idx === this.tokens.length - 1) {
+    });
+
+    function checkWaeAeTx(idx: number) {
+      if (idx === tokenList.value.length - 1) {
         return false;
       }
-      const { nodeNetworkId } = useAeSdk({ store: this.$store });
-      const contracts = DEX_CONTRACTS[nodeNetworkId.value];
-      return (
-        contracts?.wae?.includes(this.tokens[idx].contractId)
-        && this.tokens[idx + 1].isAe
-      )
-      || (
-        contracts?.wae?.includes(this.tokens[idx + 1].contractId)
-        && this.tokens[idx].isAe
+      const contracts = DEX_CONTRACTS[nodeNetworkId.value!];
+
+      return (contracts?.wae?.includes(tokenList.value[idx].contractId as Encoded.ContractAddress)
```

Comment:
I see that contractId is typed here multiple times. Couldn't you type it inside of the tokenList?

Reply:
IToken have this type`
  contractId: Encoded.ContractAddress
  | typeof AE_CONTRACT_ID
  | typeof BTC_CONTRACT_ID
`